pallets/ah-migrator/src/referenda.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::*;
+use pallet_referenda::{
+	DecidingCount, MetadataOf, ReferendumCount, ReferendumInfo, ReferendumInfoFor, TrackIdOf,
+	TrackQueue,
+};
+
+impl<T: Config> Pallet<T> {
+	pub fn do_receive_referendums(
+		referendums: Vec<(u32, ReferendumInfoOf<T, ()>)>,
+	) -> Result<(), Error<T>> {
+		log::info!(target: LOG_TARGET, "Integrating {} referendums", referendums.len());
+		Self::deposit_event(Event::ReferendumsBatchReceived { count: referendums.len() as u32 });
+		let (mut count_good, mut count_bad) = (0, 0);
+
+		for (id, referendum) in referendums {
+			match Self::do_receive_referendum(id, referendum) {
+				Ok(()) => count_good += 1,
+				Err(_) => count_bad += 1,
+			}
+		}
+
+		Self::deposit_event(Event::ReferendumsBatchProcessed { count_good, count_bad });
+		log::info!(target: LOG_TARGET, "Processed {} referendums", count_good);
+
+		Ok(())
+	}
+
+	pub fn do_receive_referendum(
+		id: u32,
+		referendum: ReferendumInfoOf<T, ()>,
+	) -> Result<(), Error<T>> {
+		log::debug!(target: LOG_TARGET, "Integrating referendum {}", id);
+
+		let referendum = if let ReferendumInfo::Ongoing(mut status) = referendum {
+			// TODO: map call and update preimage
+			// TODO: if mapping fails cancel referendum
```

Comment:
I wonder if we have to set alarms in the scheduler? The scheduler normally checks if it went into "passing" or something.  
Could be that it is not needed for big OpenGov refs since people vote on them often and that also sets the alarm.

Reply:
Right. I keep it in mind. Not sure yet how we solve it, but for sure make sure it works correctly